src/main/integrations/discord-presence/index.ts

```diff
@@ -82,172 +55,146 @@ export default class DiscordPresence implements IIntegration {
   private discordClient: DiscordClient = null;
   private enabled = false;
   private ready = false;
-  private connectionRetryTimeout: string | number | NodeJS.Timeout = null;
+  private activityDebounceTimeout: NodeJS.Timeout | null = null;
   private pauseTimeout: string | number | NodeJS.Timeout = null;
-  private endTimestamp: number | null = null;
+  private connectionRetryTimeout: string | number | NodeJS.Timeout = null;
   private stateCallback: (event: PlayerState) => void = null;
 
-  private lastTimeSeconds: number | null = null;
-  private lastDuration: number | null = null;
-
-  private lastEndTimestamp: number | null = null;
-  private lastVideoDetailsTitle: string | null = null;
-  private lastVideoDetailsAuthor: string | null = null;
-  private lastVideoDetailsId: string | null = null;
-  private lastTrackState: VideoState | null = null;
+  private videoState: VideoState | null = null;
+  private videoDetails: Partial<VideoDetails> | null = null;
+  private progress: number | null = null;
 
   private connectionRetries: number = 0;
 
-  private playerStateChanged(state: PlayerState) {
-    if (this.ready && state.videoDetails) {
-      const date = Date.now();
-      const timeSeconds = Math.floor(date / 1000);
-      const videoProgress = Math.floor(state.videoProgress);
-      const duration = state.videoDetails.durationSeconds - videoProgress;
-
-      let adjustedTimeSeconds = timeSeconds;
-
-      if (!this.lastTimeSeconds) this.lastTimeSeconds = timeSeconds;
-      if (!this.lastDuration) this.lastDuration = duration;
-      if (timeSeconds - this.lastTimeSeconds > 0) {
-        if (this.lastDuration === duration) {
-          // The time changed seconds but the duration hasn't. We use the old timestamp to prevent weird deviations
-          adjustedTimeSeconds = this.lastTimeSeconds;
-        } else {
-          this.lastDuration = duration;
-          this.lastTimeSeconds = timeSeconds;
-        }
-      } else {
-        this.lastDuration = duration;
-        this.lastTimeSeconds = timeSeconds;
+  private setActivity() {
+    if (this.activityDebounceTimeout) return;
```

Comment:
This should clear the timeout instead of returning. Currently this would only take old data when setting the activity if it were to be called multiple times within one second (as noted by how long the setTimeout is) and the latest call wouldn't be used.

Reply:
It is better like this since it will send the request to update and will update in 1 second.

But it will not delay it more than 1 second. If it cancelled it right away then it could be forever delayed. This method allows for it to be sent within 1 second and not potentially 10 minutes later, but extreme but yk

Reply:
That's not at all what it would be doing. setActivity would be creating a new timeout to replace the one you just cleared so I'm not sure where you're getting the forever delayed logic from. The point of this is so that the **latest** call to setActivity is the data used but your current logic allows setActivity to be called 100 times within 1 second but only the first call's activity data is used where as the 100th call should be the actual activity being set.

Reply:
Ah that makes sense now. I got a bit confused on the logic there. I think the function should be changed from `setActivity` to `updateActivity`. Using `set` kind of implies we're passing data to that function to be set but that's not what's logically happening.

Reply:
👍